File: snippets/lesson-7-combinelatest.cs
using System;
using System.Reactive.Subjects;
using System.Reactive.Linq;

var userStream = new BehaviorSubject<string>("Alice");
var themeStream = new BehaviorSubject<string>("Dark");

IObservable<string> viewData = userStream.CombineLatest(
    themeStream,
    (user, theme) => $"Rendering {user} in {theme} mode.");

viewData.Subscribe(Console.WriteLine);

userStream.OnNext("Bob");
themeStream.OnNext("Light");

File: snippets/lesson-7-combinelatest.ts
import { combineLatest, BehaviorSubject, Observable } from 'rxjs';
import { map } from 'rxjs/operators';

const user$ = new BehaviorSubject<string>('Alice');
const theme$ = new BehaviorSubject<string>('Dark');

const viewData$: Observable<string> = combineLatest([user$, theme$]).pipe(
  map(([user, theme]) => `Rendering ${user} in ${theme} mode.`),
);

viewData$.subscribe(console.log);

user$.next('Bob');
theme$.next('Light');